Dockerfile

```diff
@@ -1,36 +1,117 @@
-FROM amd64/ubuntu:20.04
-
-# Set-up argument defaults
+ARG ALPINE_VERSION=3.18.3
 ARG JDK_VER=11
 
-# Install supporting packages
-RUN dpkg --add-architecture i386 \
-  && apt-get update  \
-  && DEBIAN_FRONTEND=noninteractive apt-get install -y \
-  bash \
-  make \
-  openjdk-${JDK_VER}-jre-headless \
-  wine32 \
-  && rm -rf /var/lib/apt/lists/*
+ARG BASE_IMAGE=ghcr.io/stephane-d/sgdk-m68k-gcc
+ARG BASE_IMAGE_VERSION=latest
 
-# Create sgdk unprivileged user
-RUN useradd -ms /bin/sh -d /sgdk sgdk
+# Stage Zero - just init image to download files in other stages
+FROM $BASE_IMAGE:$BASE_IMAGE_VERSION as m68k-files
+
+
+FROM alpine:$ALPINE_VERSION as jdk-base
+ARG JDK_VER
+RUN apk add --no-cache openjdk${JDK_VER}
+
+
+
+# Stage One - build all tools and libs for SGDK
+FROM jdk-base as build
+RUN apk add --no-cache build-base git
 
 # Set-up SGDK
-COPY . /sgdk
-ENV GDK=/sgdk
-ENV SGDK_DOCKER=y
+WORKDIR /sgdk
+COPY . .
+ENV SGDK_PATH=/sgdk
+RUN mkdir -p $SGDK_PATH/bin
+
+# Building sjasm
+ENV SJASMEP_DIR="/tmp/sjasmep"
+RUN git clone https://github.com/istvan-v/sjasmep.git $SJASMEP_DIR
+WORKDIR $SJASMEP_DIR
+RUN make
+RUN mv $SJASMEP_DIR/sjasm $SGDK_PATH/bin/
+
+# Building bintos
+WORKDIR $SGDK_PATH/tools/bintos
+RUN gcc -O2 -s src/bintos.c -o $SGDK_PATH/bin/bintos
+
+# Building xgmtool
+WORKDIR $SGDK_PATH/tools/xgmtool
+RUN gcc -fexpensive-optimizations -Os -s src/*.c -o $SGDK_PATH/bin/xgmtool
+
+# Building apj.jar
+WORKDIR $SGDK_PATH/tools/apj/src
+RUN javac sgdk/aplib/*.java
+RUN jar cfe $SGDK_PATH/bin/apj.jar sgdk.aplib.Launcher sgdk/aplib/*.class
+
+# Building lz4w.jar
+WORKDIR $SGDK_PATH/tools/lz4w/src
+RUN javac sgdk/lz4w/*.java
+RUN jar cfe $SGDK_PATH/bin/lz4w.jar sgdk.lz4w.Launcher sgdk/lz4w/*.class
+
+# Building sizebnd.jar
+WORKDIR $SGDK_PATH/tools/sizebnd/src
+RUN javac sgdk/sizebnd/*.java
+RUN jar cfe $SGDK_PATH/bin/sizebnd.jar sgdk.sizebnd.Launcher sgdk/sizebnd/*.class
+
+# Building rescomp.jar
+WORKDIR $SGDK_PATH/tools/rescomp/src
+ENV CLASSPATH="$SGDK_PATH/bin/apj.jar:$SGDK_PATH/bin/lz4w.jar:$SGDK_PATH/tools/rescomp/src"
+RUN cp -r $SGDK_PATH/tools/commons/src/sgdk .
+RUN find . -name "*.java"
+RUN find . -name "*.java" | xargs javac
+RUN echo -e "Main-Class: sgdk.rescomp.Launcher\nClass-Path: apj.jar lz4w.jar" > Manifest.txt
+RUN jar cfm $SGDK_PATH/bin/rescomp.jar Manifest.txt  .
```

Comment:
Is there any interest in rebuilding the JAR file here instead of just copying them from the SGDK repo ? JAR should be portable.

Reply:
Sure, I'll look into it.

Reply:
This already seems to rebuild the jar from scratch.  I deleted the jar from my local repo, reran the Docker build, and everything is fine.

With this Dockerfile, I can delete all .exe, .dll, and .jar files from the repo and be perfectly happy in Linux.  Everything built from source, no Wine.  Deleting those from the repo would resolve #297, but I'm not sure if that would have negative consequences for Windows users.  Should I do it here?  In a follow-up PR?  Not at all?

Thanks!